src/events.ts

```diff
@@ -8,6 +8,7 @@ export class EventDispatcher {
 
   on(type: string, listener: Callback) {
     const { listeners } = this;
+    // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
```

Comment:
This seems like potentially an actual issue with the type annotations. We may want to set these failing rules to `warn` in the eslint config for future follow up rather than disabling with comments, which is the general strategy we take with linting upgrades that surface new errors

Reply:
I think that's a good strategy overall and will implement, but for this line in particular shouldn't it be ignored? 

My understanding of this line is that `listeners` is a map of `String: Set`. Since the key represented by the var `type` might not be in the map, it could come back as undefined, in which case we do a conditional assignment to a new `Set` (while not assigning if it exists). 

I don't see this line as an issue, but maybe I'm missing something?

Reply:
I think this is complaining because `listeners` is defined as a `Record<string, Set<...>>`, which says "every string field is a `Set`." This lint is telling you "if every string field is a set, then you don't need to conditionally assign." I think this this issue can be fixed by setting the type signature to say "every string field is `Set` or is `undefined`" using `Partial`

```ts
listeners: Partial<Record<string, Set<Callback>>> = {};
```